management/src/main/scala/akka/management/AkkaManagementSettings.scala

```diff
@@ -49,7 +49,17 @@ final class AkkaManagementSettings(val config: Config) {
     val BasePath: Option[String] =
       Option(cc.getString("base-path")).flatMap(it ⇒ if (it.trim == "") None else Some(it))
 
-    val RouteProviders: immutable.Seq[String] = cc.getStringList("route-providers").asScala.toList
+    val RouteProviders: immutable.Seq[NamedRouteProvider] = {
+      cc.getConfig("routes")
+        .root
+        .unwrapped
+        .asScala
+        .map {
+          case (name, value) => NamedRouteProvider(name, value.toString)
+        }
+        .toList
+        .filter(_.fullyQualifiedClassName != "")
```

Comment:
Instead of `map` and `filter`, we could use a `collect`.

```
def valideFQCN(value: String) = {
  value != null || value.trim.nonEmpty || value != "disabled"
}

...

.collect {
  case (name, value) if valideFQCN(value.toString) => NamedRouteProvider(name, value.toString)
} 
```